### ISSUE TITLE
[Feature] Calculate anchor boxes by kmeans and GA

### ISSUE THREAD

**Issue by rehohoho** [whizzmobility/models#69]
# Prerequisites

Please answer the following question for yourself before submitting an issue.

- [x] I checked to make sure that this feature has not been requested already.

## 1. The entire URL of the file you are using

https://github.com/tensorflow/models/tree/master/official/vision/beta

## 2. Describe the feature you request

Calculate the anchor boxes to use for one-shot detector strategies according to labels, using
- kmeans
- genetic algorithm

Some understanding on the boxes:
https://towardsdatascience.com/anchor-boxes-the-key-to-quality-object-detection-ddf9d612d4f9
pjreddie/darknet#568

Some implementations on generating boxes using clustering studies:
official darknet: https://github.com/AlexeyAB/darknet/blob/master/scripts/gen_anchors.py
ultralytics: https://github.com/ultralytics/yolov5/blob/1093b1d765fd2d1d603affdbb1473554b396bf1c/utils/autoanchor.py
ultralytics: ultralytics/yolov5#3546
keras: https://github.com/experiencor/keras-yolo3/blob/master/gen_anchors.py

## 3. Additional context

Add any other context about the feature request here.

## 4. Are you willing to contribute it? (Yes or No)

Yes

**Comment by rehohoho** [comment]
Alternatively use YOLOX, that supercedes YOLOv4 and YOLOv5 which does not use anchors.
https://github.com/Megvii-BaseDetection/YOLOX
https://arxiv.org/pdf/2107.08430v1.pdf

Improvements:
- Anchor-free
- Advanced label assignment strategy
- End-to-end (no NMS)
